warehouse/admin.go

```diff
@@ -77,30 +88,34 @@ func (wh *WarehouseAdmin) Query(s QueryInput, reply *warehouseutils.QueryResult)
 	return err
 }
 
-func (wh *WarehouseAdmin) QueryWhUploads(uploadsReq UploadsReqT, reply *[]byte) error {
-	uploadsReq.API = UploadAPI
-	res, err := uploadsReq.GetWhUploads()
-	if err != nil {
-		return err
+// Test the underlying warehouse
+func (wh *WarehouseAdmin) Test(s TestInput, reply *TestResponse) (err error) {
+	if strings.TrimSpace(s.DestID) == "" {
+		return errors.New("please specify the destination ID to query the warehouse")
```

Comment:
I see you use named return arguments extensively in codebase. I was thinking about them a lot and it has good benefits but one thing it causes is mixed return statements and implicit shadowing of variables.

Let's discuss this 1-1 when you have time ?